client/v2/autocli/query.go

```diff
@@ -19,12 +21,17 @@ import (
 func (b *Builder) BuildQueryCommand(moduleOptions map[string]*autocliv1.ModuleOptions, customCmds map[string]*cobra.Command) (*cobra.Command, error) {
 	queryCmd := topLevelCmd("query", "Querying subcommands")
 	queryCmd.Aliases = []string{"q"}
-	for moduleName, modOpts := range moduleOptions {
+
+	// Ensure that building module query commands is in a deterministic pattern.
+	moduleNames := maps.Keys(moduleOptions)
+	sort.Strings(moduleNames)
+	for _, moduleName := range moduleNames {
```

Comment:
Is it documented somewhere that BuildQueryCommand should iterate over module names deterministically? If not, is it necessary to do so? If so, is there a test which enforces this invariant? And isn't manual sorting of keys somewhat error prone? Should there rather be a package/function that ensures deterministic order without requiring explicit sorting at the callsite?